express-zod-api/src/sse.ts

```diff
@@ -55,9 +57,18 @@ export const ensureStream = (response: Response) =>
 
 export const makeMiddleware = <E extends EventsMap>(events: E) =>
   new Middleware({
-    handler: async ({ response }): Promise<Emitter<E>> =>
-      setTimeout(() => ensureStream(response), headersTimeout) && {
+    handler: async ({ request, response }): Promise<Emitter<E>> => {
+      const controller = new AbortController();
+
+      request.on("close", () => {
```

Comment:
@jakub-msqt , should we only run this once?
Thus, we'd also avoid open listeners.

```suggestion
      request.once("close", () => {
```

Reply:
Sure, I think it makes more sense 👍